src/librustc_span/lib.rs

```diff
@@ -923,11 +957,13 @@ pub struct SourceFile {
     pub normalized_pos: Vec<NormalizedPos>,
     /// A hash of the filename, used for speeding up hashing in incremental compilation.
     pub name_hash: u128,
+    /// Cryptographic hash of on-disk bytes for debug info.
+    pub src_file_checksum: Option<SourceFileChecksum>,
```

Comment:
See `pub src_hash: u128,` - we already have a hash, it's just not MD5 or SHA-1 (it's probably Blake?). We should maybe consolidate the two so we're not computing several hashes of the same file.

cc @michaelwoerister @petrochenkov

Reply:
`src_hash` is the 128 bit version of SipHash. It would not consider that cryptographically secure (although both MD5 and SHA-1 are not much better at this point, right?)

Reply:
Oh, we're not using BLAKE2? (anymore?) Then switching to md5 seems fine with me.

Reply:
No, we moved away from BLAKE2 because it was unnecessarily slow and we don't need the strong guarantees it gives: https://github.com/rust-lang/rust/issues/41215